### PR TITLE
Add M5Stack Tough

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Add M5Stack Tough

## Motivation and Context

- Add M5Stack Tough

## How Has This Been Tested?

Need to be tested.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by josesimoes** [comment]
@Ellerbach I've refactor a bit the Core2ToughCommon class. Namely:
- Properties: had to wrap some to make them available only when they exist.
- Constructor: reorder the code, adjusted some bits that weren't following M5Stack values and add the missing bits for the Tough.

Also some refactoring with the Screen class:
- Some renames
- Rework the Enable code to deal only with the backlight. As it was it was disabling the power for all peripherals (they all share the same power rail).
- Rename the luminosity to brightness (for consistency with common nomenclature).

**Comment by josesimoes** [comment]
@Ellerbach we need to rethink the touch implementation too. The Tough doesn't have "buttons" as the Core2. 
So it' up to the application layer to setup the touch "objects" on the screen and deal with the touch events.
Suggest that we remove the touch for now until we figure it out.

File: M5StackCommon/Core2ToughCommon.cs [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+                {
+                    Rtc.SetSystemTime(dt);
+                }
+            }
+            catch (Exception)
+            {
+
+                if (sysDtcore.Year < 2021)
+                {
+                    dt = new DateTime(2021, 11, 03, 12, 00, 00);
```
Comment (Ellerbach) [inline M5StackCommon/Core2ToughCommon.cs:419]: maybe increase to today? :-)

File: M5StackCommon/Screen.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            set
+            {
+                // For M5Core2 and M5Tough, values from 2.5 to 3V are working fine
+                // 2.5 V = dark, 3.0 V full luminosity
+                _brightness = (byte)(value > 100 ? 100 : value);
+                var backLightVoltage = ElectricPotential.FromVolts(_brightness * 0.5 / 100.0 + 2.5);
+#if M5CORE2
+                _power.LDO3OutputVoltage = backLightVoltage;
+#elif TOUGH
+            _power.LDO3OutputVoltage = backLightVoltage;
```
Comment (Ellerbach) [inline M5StackCommon/Screen.cs:113]: missing few spaces here as well

File: Tests/ToughTestApp/Program.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+using System.Diagnostics;
+using System.Threading;
+using Console = nanoFramework.M5Stack.Console;
+
+Tough.InitializeScreen();
+
+Debug.WriteLine("Hello from Tough!");
+
+Console.WriteLine("Hello from Tough!");
+
```
Comment (Ellerbach) [inline Tests/ToughTestApp/Program.cs:18]: minor: extra line

File: M5StackCommon/Core2ToughCommon.cs [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+            
+            // Sets the LCD backlight voltage to 3V
+            _power.LDO3OutputVoltage = ElectricPotential.FromVolts(3.0);
+
+              // GPIO1 is reset for Touch controller
+            _power.Gpio1Behavior = Gpio12Behavior.MnosLeakOpenOutput;
+
+#endif
+
+            // Setup buttons
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply M5StackCommon/Core2ToughCommon.cs:378]: Fine with me... I know the guy who coded this, I'll bring this to his attention... 😝

File: M5StackCommon/Core2ToughCommon.cs (hunk 1 above)
Comment (josesimoes) [reply M5StackCommon/Core2ToughCommon.cs:419]: To be honest I didn't even looked at this line... Done!

**Review by Ellerbach — COMMENTED** [review]
Only thing missing now if adjusting the property to get the GpioController. So far it's not done! IT's a common one to the rest. So need to be adjusted ;-)

File: nanoFramework.M5Core/M5Core.cs [hunk 5]
```diff
@@ -151,9 +151,6 @@ static M5Core()
             _power.ChargingCutOffCurrent = ChargingCutOffCurrent.C500mA;
             _power.ChargingCuttOffVoltage = ChargingCutOffVoltage.V4_2;
 
-            // Setup buttons
-            _gpio = new();
```
Comment (Ellerbach) [inline nanoFramework.M5Core/M5Core.cs:155]: This must stay here because of the buttons

**Comment by josesimoes** [comment]
Ignoring Sonarcloud duplication error.

File: nanoFramework.Fire/Fire.cs [hunk 6]
```diff
@@ -166,9 +166,6 @@ static Fire()
             _power.ChargingCutOffCurrent = ChargingCutOffCurrent.C500mA;
             _power.ChargingCuttOffVoltage = ChargingCutOffVoltage.V4_2;
 
-            // Setup buttons
-            _gpio = new();
```
Comment (Ellerbach) [inline nanoFramework.Fire/Fire.cs:170]: This must stay here because of the buttons

**Review by Ellerbach — COMMENTED** [review]
Looks good to me! Can't approve as I'm the original author! So feel free to approve and merge!

**Pull request merged**